tests/test_mongodb_index.py

```diff
@@ -0,0 +1,295 @@
+
+            # # max 32 fields in a compound index (NO PLANS TO SUPPORT THIS IN THE TAP)
+            # client["simple_db"]["simple_coll_1"].create_index([
+            #     ('string_field', pymongo.ASCENDING), ('string_field_02', pymongo.ASCENDING),
+            #     ('string_field_03', pymongo.ASCENDING), ('string_field_04', pymongo.ASCENDING),
+            #     ('string_field_05', pymongo.ASCENDING), ('string_field_06', pymongo.ASCENDING),
+            #     ('string_field_07', pymongo.ASCENDING), ('string_field_08', pymongo.ASCENDING),
+            #     ('string_field_09', pymongo.ASCENDING), ('string_field_10', pymongo.ASCENDING),
+            #     ('string_field_11', pymongo.ASCENDING), ('string_field_12', pymongo.ASCENDING),
+            #     ('string_field_13', pymongo.ASCENDING), ('string_field_14', pymongo.ASCENDING),
+            #     ('string_field_15', pymongo.ASCENDING), ('string_field_16', pymongo.ASCENDING),
+            #     ('string_field_17', pymongo.ASCENDING), ('string_field_18', pymongo.ASCENDING),
+            #     ('string_field_19', pymongo.ASCENDING), ('string_field_20', pymongo.ASCENDING),
+            #     ('string_field_21', pymongo.ASCENDING), ('string_field_22', pymongo.ASCENDING),
+            #     ('string_field_23', pymongo.ASCENDING), ('string_field_24', pymongo.ASCENDING),
+            #     ('string_field_25', pymongo.ASCENDING), ('string_field_26', pymongo.ASCENDING),
+            #     ('string_field_27', pymongo.ASCENDING), ('string_field_28', pymongo.ASCENDING),
+            #     ('string_field_29', pymongo.ASCENDING), ('string_field_30', pymongo.ASCENDING),
+            #     ('string_field_31', pymongo.ASCENDING), ('string_field_32', pymongo.ASCENDING)])
+
+            self.index_info = client["simple_db"]["simple_coll_1"].index_information()
+
+    def expected_check_streams(self):
+        return {
+            'simple_db-simple_coll_1',
+            'simple_db-simple_coll_2',
+        }
+
+    def expected_pks(self):
+        return {
+            'simple_coll_1': {'_id'},
+            'simple_coll_2': {'_id'},
+        }
+
+    def expected_row_counts(self):
+        return {
+            'simple_coll_1': 50,
+            'simple_coll_2': 100,
+        }
+
+    def expected_sync_streams(self):
+        return {
+            'simple_coll_1',
+            'simple_coll_2'
+        }
+
+    def name(self):
+        return "tap_tester_mongodb_oplog"
+
+    def tap_name(self):
+        return "tap-mongodb"
+
+    def get_type(self):
+        return "platform.mongodb"
+
+    def get_credentials(self):
+        return {'password': os.getenv('TAP_MONGODB_PASSWORD')}
+
+    def get_properties(self):
+        return {'host' : os.getenv('TAP_MONGODB_HOST'),
+                'port' : os.getenv('TAP_MONGODB_PORT'),
+                'user' : os.getenv('TAP_MONGODB_USER'),
+                'database' : os.getenv('TAP_MONGODB_DBNAME')
+        }
+
+    def expected_string_fields(self):
```

Comment:
This set could be build out programmatically. I generally prefer to do things that way because it helps keep typos out and increases maintainability. The only time I think we should avoid generating data programmatically is if it drastically hurts readability or decreases our confidence in the test.

Reply:
Agreed, updated.